exana/tracking/fields.py

```diff
@@ -454,13 +501,20 @@ def spatial_rate_map_1d(x, t, sptr,
     # correct for shifting of map since digitize returns values at right edges
     spike_pos = spike_pos[1:]
     time_pos = time_pos[1:]
+
+    if convolve and convolve_spikes_and_time:
+        from astropy.convolution import Gaussian1DKernel, convolve_fft
+        csize = (track_len / binsize) * smoothing
+        kernel = Gaussian1DKernel(csize)
+        spike_pos = convolve_fft(spike_pos, kernel)  
+        time_pos  = convolve_fft(time_pos, kernel)  
     with np.errstate(divide='ignore', invalid='ignore'):
         rate = np.divide(spike_pos, time_pos)
-    if convolve:
+    if convolve and not convolve_spikes_and_time:
         rate[np.isnan(rate)] = 0.  # for convolution
-        from astropy.convolution import Gaussian2DKernel, convolve_fft
+        from astropy.convolution import Gaussian1DKernel, convolve_fft
```

Comment:
Did you find the reason why this was changed from 2D to 1D @halvarsu ?

Reply:
I think it makes sense, as it is the 1D-spatial rate map function :)

Reply:
True :) I thought it was used in the 2D case as well, but I probably just remember wrong